.github/workflows/snap.yml

```diff
@@ -0,0 +1,52 @@
+name: Snap CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "packages/**"
+
+jobs:
+  build-lint-test:
+    name: Build, Lint, and Test
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        node-version: [16.x, 18.x]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Get Yarn cache directory
+        run: echo "::set-output name=YARN_CACHE_DIR::$(yarn cache dir)"
+        id: yarn-cache-dir
+      - name: Get Yarn version
+        run: echo "::set-output name=YARN_VERSION::$(yarn --version)"
+        id: yarn-version
+      - name: Cache yarn dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.yarn-cache-dir.outputs.YARN_CACHE_DIR }}
+          key: yarn-cache-${{ runner.os }}-${{ steps.yarn-version.outputs.YARN_VERSION }}-${{ hashFiles('yarn.lock') }}
```

Comment:
You should be able to do this more easily with [actions/setup-node@v3](https://github.com/actions/setup-node#caching-global-packages-data):

```yaml
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v3
        with:
          node-version: ${{ matrix.node-version }}
          cache: yarn # <-- add this
```

[Example from BLS Wallet extension workflow](https://github.com/web3well/bls-wallet/blob/main/.github/workflows/extension.yml#L26-L30)

Reply:
done